PR_projekt1/Algorytm.cpp

```diff
@@ -2,161 +2,191 @@
 #include <omp.h>
 #include <cstdio>
 #include <cmath>
+#include <iostream>
+#include <algorithm>
 
-void printResult(std::vector<int> result) {
-	printf("znaleziono %d liczb pierwszych w zadanym przedziale", result.size());
-	for (int i = 0; i < result.size(); i++) {
-		if (i % 10 == 0) {
-			printf("\n");
+void printResult(std::vector<int> primes) {
+	std::cout << "znaleziono " << primes.size() << " liczb pierwszych w zadanym przedziale" << std::endl;
+	
+	for (int i = 0; i < primes.size(); i++) {
+		std::cout << primes[i] << " ";
+		if (i > 0 && i % 10 == 0) {
+			std::cout << std::endl;
 		}
-		printf("%d ", result[i]);
 	}
 }
 
 //TODO test efektywno�ci czy wyznacza� liczby pierwsze jako dzielnika - kiedy op�acalne
 
-std::vector<int> getFactorsFromEratostenes(int maxFactor) {
-	std::vector<int> factors;
-	bool* isComposite = new bool[maxFactor + 1]();
+std::vector<int> getPrimeFactorsFromEratostenes(int maxFactor) {
```

Comment:
z ciekawości możemy zostawić obie wersje (z tablicą i wektorem) i zobaczyć czy jest jakaś różnica czasowa

Reply:
Nie będzie z racji braku insercji, wektor ma już ustalony rozmiar. Jedynie zawiera więcej pamięci. Wektor będzie co najwyżej tak szybki jak tablica, dlatego są to znikome wartości, nie mają żadnego wpływu na wydajność.